### PR TITLE
Draft: Deploy a Spring Boot app on your own server in less than 15 minutes

### PR DESCRIPTION
**Pull request by pascalw**
This is an experiment to find a nice workflow for collaborative Markdown editing. Using Google Drive to get feedback on writing is nice, but Google Drive sucks at Markdown. This specific post has quite a bit of Markdown formatting making it very hard to read in Drive.

Advantages of using Github for draft reviews are:

* You can conveniently have Github interpret the Markdown, effectively providing you with a real view of how the post will look on the website
* Even the diff view is much more readable in Github because provide some Markdown syntax highlighting
* We can store drafts in the repo, which might be a good idea in general
* Can use embedded images in draft

Please use the regular Pull Request comments workflow to provide feedback on the post and also let me know what you think about using Github for draft reviews.

### CODE REVIEW

**Comment by matthijsgroen** [comment]
Cool! I was experimenting with markdown editors on drive, but they sucked in connectivity, responsiveness and you could not do realtime collaboration (Tried StackEdit). I like this approach!

File: drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+       DOKKU_APP_RESTORE: 1
+=====> Application deployed:
+       http://my-app.example.org
+```
+
+Open your browser at `http://my-app.dokku.example.org/` and see a nice random quote!
+
+## Closing thoughts
+
+Dokku takes a bit of one time setup but once you have your Dokku instance ready deploying new applications will be a breeze. Do note that Dokku was designed for simple single server setups. If you need fancy stuff like clustering, access control etc you might want to look at one of the more comprehensive projects in this space like [Flynn](https://flynn.io) or [Deis](http://deis.io).
```
Comment (matthijsgroen) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: Run - on - sentence

File: drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+Even though Dokku is quite simple we've only scratched the surface of what Dokku can do. Now you have a fully functional Dokku instance go ahead and explore the [Dokku docs](http://dokku.viewdocs.io/dokku/installation/) to see what more is possible.
```
Comment (matthijsgroen) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: simple, we've

Comment (matthijsgroen) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: instance. Go ahead ... possible!

**Comment by pascalw** [comment]
Thanks for your feedback @matthijsgroen!

What I don't really like about reviewing content like this is that it's a bit cumbersome to suggest changes. On Google Docs you can exactly highlight the content your commenting on.

There are a couple of tools in this space like https://draftin.com but all of them seem to have weird quirks (for example draftin is terrible at rendering code snippets) or the writer and reviewers all need to signup etc.

**Comment by matthijsgroen** [comment]
Yes I went to `files -> the .md file -> show file` for reading, but you can't add a comment to a line then, so then you need to switch to the code view back to add a comment to the correct line.

**Comment by ruisalgado** [comment]
the Dokku website has [further installation notes](http://dokku.viewdocs.io/dokku/installation/) for digital ocean, linode & azure. It's probably worth to link to it.

File: drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+       http://my-app.example.org
+```
+
+Open your browser at `http://my-app.dokku.example.org/` and see a nice random quote!
+
+## Closing thoughts
+
+Dokku takes a bit of one time setup but once you have your Dokku instance ready, deploying new applications will be a breeze. Do note that Dokku was designed for simple single server setups. If you need fancy stuff like clustering, access control etc you might want to look at one of the more comprehensive projects in this space like [Flynn](https://flynn.io) or [Deis](http://deis.io).
+
+Even though Dokku is quite simple, we've only scratched the surface of what Dokku can do. Now you have a fully functional Dokku instance go ahead and explore the [Dokku docs](http://dokku.viewdocs.io/dokku/installation/) to see what more is possible!
```
Comment (ruisalgado) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: ... scratched the surface of what **it** can do.

**Comment by ruisalgado** [comment]
There are a lot of analogies made with Heroku, which I'm not sure is very popular with Java devs. So maybe it'd be useful to add a bit more context or make the comparison with other PaaS like Cloudfoundry or Openshift

**Comment by pascalw** [comment]
Reduced analogies to Heroku and mentioned a few other PaaSs.

I've also replaced the deployment output wall of text with an animated gif.

File: drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+Dokku uses a Heroku-style `git push to deploy` model, so we'll need to add a Git remote to our repo to point to Dokku's Git repo. Replace `dokku.example.org` with your actual Dokku hostname.
+
+```sh
+$ git remote add dokku dokku@dokku.example.org:my-app
+```
+
+And finally push our app to Dokku:
+
+```sh
+$ git push dokku
```
Comment (Hacklor) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: I actually had to use a `git push dokku master`.

Maybe other readers will run into the same, cus of:
dokku-boot-demo 9h master / git push master dokku                                                                           ruby-2.1.2
error: src refspec dokku does not match any.
error: failed to push some refs to 'master'

**Comment by Hacklor** [comment]
Could follow the blog well, although I worked with IP address instead of host name.

Looking back I didn't actually set up the database url. Picked that up automatically.

File: drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md (hunk 4 above)
Comment (pascalw) [inline drafts/deploy-spring-boot-app-in-15-minutes/deploy-spring-boot-app-in-15-minutes.md]: Ah yes good point. `git push dokku` might not working depending on your `push.default` git config. Will change it to `git push dokku master` that'll always work.

**Comment by pascalw** [comment]
> Looking back I didn't actually set up the database url. Picked that up automatically.

I'm not sure I understand what you mean. Did you skip on of the database related steps?

**Comment by pascalw** [comment]
Merging as the post has been published. Thanks for all the feedback!

**Pull request merged**